templates/master/00-master/baremetal/files/baremetal-haproxy.yaml

```diff
@@ -72,15 +74,35 @@ contents:
         imagePullPolicy: IfNotPresent
       containers:
       - name: haproxy
-        image: docker.io/library/haproxy:latest
-        args:
-        - "-W"
-        - "-db"
-        - "-S"
-        - "/var/run/haproxy/haproxy-master.sock,level,admin"
-        - "-f"
-        - "/etc/haproxy/haproxy.cfg"
+        image: openshift/origin-haproxy-router:latest
+        command:
+        - "/bin/bash"
+        - "-c"
+        - |
+          #/bin/bash
+          set -ex
+          function reload_haproxy
```

Comment:
Didn't Russell suggest locking on the file or using some unix socket?

How about we make it so that we can preserve functionality on the controller side doing the same that the unix socket does in 1.9. It would look similar to the below code:

````
#!/bin/bash

declare -r haproxy_sock="/var/run/haproxy/haproxy-master.sock
if [ -S "$haproxy_sock" ]; then
    rm "$haproxy_sock"
fi

function handle() {
    declare -r shell_pid="$1"
    declare -r haproxy_pid="$2"

    read -r cmd;
    while [[ "$cmd" != "quit" ]]; do
        if [[ "$cmd" == "reload" ]]; then
            kill -s SIGUSR2 "$haproxy_pid"
        fi
        read -r cmd
    done
    echo -n "quitting..."
    kill -- -"$shell_pid"
}

ss_pid=${BASHPID}

/usr/sbin/haproxy -W -db -f /etc/haproxy/haproxy.cfg &
declare -r haproxy_pid="$!"

nc -lkU "$haproxy_sock" | handle "$ss_pid" "$haproxy_pid"
````